src/script/cond_stack.zig

```diff
@@ -0,0 +1,153 @@
+const std = @import("std");
+const testing = std.testing;
+const Allocator = std.mem.Allocator;
+
+/// Errors that can occur during stack operations
+pub const ConditionalStackError = error{
+    EmptyConditionalStack,
+    InvalidCondition,
+    DisabledOpcode,
+    UnbalancedConditional,
+};
+
+/// ConditionalStack represents a stack of conditional values used in script execution
+///
+/// This struct implements a stack data structure using a dynamic array.
+/// It manages conditional values (0, 1, 2) for branching in script execution.
+pub const ConditionalStack = struct {
+    const Self = @This();
+
+    /// Dynamic array to store conditional values
+    stack: std.ArrayList(u8),
+    /// Memory allocator used for managing stack storage
+    allocator: Allocator,
+
+    /// Initialize a new ConditionalStack
+    ///
+    /// Args:
+    ///     allocator: Memory allocator for managing stack storage
+    ///
+    /// Returns: A new ConditionalStack instance
+    pub fn init(allocator: Allocator) Self {
+        return .{
+            .stack = std.ArrayList(u8).init(allocator),
+            .allocator = allocator,
+        };
+    }
+
+    /// Deallocate all resources used by the ConditionalStack
+    pub fn deinit(self: *Self) void {
+        self.stack.deinit();
+    }
+
+    /// Push a conditional value onto the stack
+    ///
+    /// Args:
+    ///     value: The conditional value to push (typically 0, 1, or 2)
```

Comment:
Can it be something else than 0, 1 or 2? You comment suggest it can

Reply:
Also can you define an enum for more readability
```
	OpCondFalse = 0
	OpCondTrue  = 1
	OpCondSkip  = 2
```

Reply:
> Can it be something else than 0, 1 or 2? You comment suggest it can

Reading other implementations it looks like it cannot. So please change your comment to not mean that it could

Reply:
And ofcourse use those OpCond enum in your tests rather than 1, 2, 3

Reply:
I believe it can, here is a go implementation where it does

https://github.com/btcsuite/btcd/blob/b161cd6a199b4e35acec66afc5aad221f05fe1e3/txscript/opcode.go#L972